packages/itwinui-react/src/core/ComboBox/ComboBox.tsx

```diff
@@ -536,67 +555,96 @@ export const ComboBox = React.forwardRef(
 
     return (
       <ComboBoxRefsContext.Provider
-        value={{ inputRef, menuRef, optionsExtraInfoRef }}
+        value={{ inputRef, menuRef, optionsExtraInfo }}
       >
-        <ComboBoxActionContext.Provider value={dispatch}>
-          <ComboBoxStateContext.Provider
-            value={{
-              id,
-              isOpen,
-              focusedIndex,
-              onClickHandler,
-              enableVirtualization,
-              filteredOptions,
-              getMenuItem,
-              multiple,
-              popover,
-              show,
-              hide,
-            }}
+        <ComboBoxStateContext.Provider
+          value={{
+            id,
+            isOpen,
+            focusedIndex,
+            setFocusedIndex,
+            onClickHandler,
+            enableVirtualization,
+            filteredOptions,
+            getMenuItem,
+            multiple,
+            popover,
+            show,
+            hide,
+          }}
+        >
+          <ComboBoxInputContainer
+            ref={forwardedRef}
+            disabled={inputProps?.disabled}
+            {...rest}
           >
-            <ComboBoxInputContainer
-              ref={forwardedRef}
-              disabled={inputProps?.disabled}
-              {...rest}
-            >
-              <>
-                <ComboBoxInput
-                  value={inputValue}
-                  disabled={inputProps?.disabled}
-                  {...inputProps}
-                  onChange={handleOnInput}
-                  selectTags={
-                    isMultipleEnabled(selected, multiple)
-                      ? selected.map((index) => {
-                          const item = optionsRef.current[index];
+            <>
+              <ComboBoxInput
+                value={inputValue}
+                disabled={inputProps?.disabled}
+                {...inputProps}
+                onChange={handleOnInput}
+                selectTags={
+                  isMultipleEnabled(selectedIndexes, multiple)
+                    ? (selectedIndexes
+                        ?.map((index) => {
+                          const item: SelectOption<T> | undefined =
+                            optionsRef.current[index];
+
+                          if (item == null) {
```

Comment:
Why would any of the items be null?

Reply:
The problem might be that we were using `optionsRef` during rendering (discouraged in [react docs](https://react.dev/learn/referencing-values-with-refs#differences-between-refs-and-state) probably to avoid this exact issue).

Can we directly use `options[index]` instead?